test/gtest/ucp/test_ucp_wakeup.cc

```diff
@@ -105,3 +211,59 @@ UCS_TEST_P(test_ucp_wakeup, signal)
 }
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_wakeup)
+
+class test_ucp_tx_wakeup : public test_ucp_wakeup
+{
+public:
+    static ucp_worker_params_t get_worker_params() {
+        ucp_worker_params_t params = test_ucp_wakeup::get_worker_params();
+        params.field_mask |= UCP_WORKER_PARAM_FIELD_EVENTS;
+        params.events = UCP_WAKEUP_TX_COMPLETION;
```

Comment:
generate_test_params_variant() is called during gtest startup to create all possible test parameters, it's possible to use it to generate 2 variants for the existing parameters: RX and TX. then use GetParam() during the test to figure on which is it for each test instantiation.